client-sdk/ts-web/core/src/hdkey.ts

```diff
@@ -0,0 +1,234 @@
+        public readonly c: Uint8Array,
+    ) {
+        this.secret = concat(kL, kR);
+    }
+
+    /**
+     * Returns the root node for the given mnemonic
+     * @param mnemonic Bip39 Mnemonic
+     * @returns Root node
+     */
+    public static async fromMnemonic(mnemonic: string): Promise<HdKey> {
+        const seed = await mnemonicToSeed(mnemonic);
+        return HdKey.fromSeed(seed.slice(0, 32));
+    }
+
+    /**
+     * Checks if the given bip39 mnemonic is valid for Ed25519 derivation
+     * If it is not, you must generate another one.
+     * @param mnemonic
+     */
+    public static async generateValidMnemonic(strength = 256): Promise<string> {
+        let mnemonic = generateMnemonic(strength);
+        while (!(await HdKey.isValidMnemonic(mnemonic))) {
+            mnemonic = generateMnemonic(strength);
+        }
+
+        return mnemonic;
+    }
+
+    /**
+     * Checks if the given bip39 mnemonic is valid for Ed25519 derivation
+     * If it is not, you must generate another one.
+     * @param mnemonic
+     */
+    public static async isValidMnemonic(mnemonic: string): Promise<boolean> {
+        const seed = await mnemonicToSeed(mnemonic);
+        const hash = new Uint8Array(sha512.arrayBuffer(seed.slice(0, 32)));
+
+        // If the third highest bit of the lasst byte of kL is not zero, discard k
+        return (hash[31] & 0b00100000) === 0;
+    }
+
+    /**
+     * Returns the root node for the given 32 bytes seed
+     * @param seed 32 bytes seed
+     * @returns Root node
+     */
+    public static fromSeed(seed: Uint8Array): HdKey {
+        if (seed.length != 32) throw 'Seed must be 32 bytes (a Uint8Array of size 32)';
+
+        const k = new Uint8Array(sha512.arrayBuffer(seed));
+        const kL = k.slice(0, 32),
+            kR = k.slice(32);
+
+        //  If the third highest bit of the last byte of kL is not zero, discard k
+        if (kL[31] & 0b00100000) {
+            return null;
+        }
+
+        // clear lowest three bits of the first byte
+        // clear highest bit of the last byte
+        // set second highest bit of the last byte
+        kL[0] &= 248;
+        kL[31] &= 127;
+        kL[31] |= 64;
+
+        // Get the public key
+        const A = new Uint8Array(HdKey.eddsa.encodePoint(HdKey.bp.mul(bn(kL, 10, 'le'))));
+
+        const chaincodePayload = concat(new Uint8Array([1]), seed);
+        const c = new Uint8Array(sha256.arrayBuffer(chaincodePayload));
+
+        return new HdKey(kL, kR, A, c);
+    }
+
+    /**
+     * Derive a new key at the specified derivation path from the current key
+     * @param path Derivation path, using ' for hardened indexes
+     * @returns Child node at path
+     */
+    public derivePath(path: string): HdKey {
+        let node: HdKey = this;
+        const chain = path.split('/');
+
+        for (let i = 0; i < chain.length; i++) {
+            let index;
+            if (!chain[i]) continue;
+            if (chain[i].endsWith("'"))
```

Comment:
I suggest using explicit braces everywhere even if the body is just a single line.

Is there a lint that we can enable for this?

Reply:
https://github.com/prettier/prettier/issues/7659#issuecomment-590004436 ): I just searched, and it looks like prettier doesn't want to include this kind of thing in the scope of their tool. we should probably go down the road of eslint then

Reply:
Prettier does not support this rule, I'll change it right away though